### ISSUE TITLE
Check before overwriting existing file

### ISSUE THREAD

**Issue by christophdecker** [ilastik/ilastik#937]
Applying Pixel Classification to "my_dataset.h5" produces "my_dataset_export.h5". Using these two files as the input for another workflow such as Automated Tracking works fine, but once you export the results of this workflow, it will write them to disk with the default name "my_dataset_export.h5" and thus overwrite the previously produced Pixel Classification result without telling you. 

Of course you could check for this in the export settings before exporting, but it's still not so easy to catch if it didn't happen to you before, because you don't specify the exact export file name. In case you needed the Pixel Classification results you'll have to reproduce them which might take hours or even days (as is the case for me) - and here you also have to make sure to change the default export file name to not overwrite your Tracking/Object Classification/whatsoever results.

Long story short: Before writing any file, ilastik should definitely check if a file with this name already exists and prompt the user to either provide a different name or confirm to overwrite the old one.
